### PR TITLE
feat: add public access for providerS3

### PR DESCRIPTION
**Pull request by LukasMod**
### Summary
Adds public access support to the S3 provider. Adds additional config flag to allow provider to configure a custom signer that doesn't sign requests, enabling access to public S3 buckets without authentication. This supports public buckets and simplifies configuration when credentials aren't needed.

### Test Plan
Unit tests that verifies:
- S3Client is configured with a custom signer and empty credentials when no auth is provided
- The signer function returns requests unchanged (no signing)
- The provider works correctly for downloading artifacts (mock)

Manual test with public file:  https://ad-hoc-expensify-cash.s3.amazonaws.com/rock-artifacts/rock-android-developmentDebug-bc955486ab50749733e23713a2ef32fefa1c7342.zip

---

> [!NOTE]
> Introduces a `publicAccess` flag to the S3 provider to enable unauthenticated access to public buckets with a no-op signer, updates error handling on download, and adds docs/tests.
> 
> - **Provider (`packages/provider-s3/src/lib/providerS3.ts`)**
>   - Add `publicAccess` to `ProviderConfig`.
>   - When `publicAccess` is true, configure S3 client with no-op `signer` and empty `credentials`.
>   - Improve `download()` to surface a clearer error message when using public access.
> - **Tests (`packages/provider-s3/src/__tests__/providerS3.test.ts`)**
>   - Add test ensuring S3 client uses no-op signer/empty credentials and `download` works under `publicAccess`.
> - **Docs**
>   - README and site config docs: document `publicAccess` option and add a public S3 usage example.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 8b568f638fd8d6c71470f90c582dd0a7f3e4b32c. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: packages/provider-s3/src/lib/providerS3.ts [hunk 1]
```diff
@@ -104,6 +104,15 @@ export class S3BuildCache implements RemoteBuildCache {
...
       s3Config.credentials = fromIni({ profile: config.profile });
+    } else {
+      // Fallback to public access
+      s3Config.signer = {
+        sign: async (request) => request,
+      };
+      s3Config.credentials = {
+        accessKeyId: '',
+        secretAccessKey: '',
+      };
```
[the remark this answers is not in the document]
Comment (LukasMod) [reply packages/provider-s3/src/lib/providerS3.ts]: updated with `publicAccess`
